editor/components/inserter/menu.js

```diff
@@ -255,6 +256,14 @@ export class InserterMenu extends Component {
 	}
 
 	renderCategories( visibleBlocksByCategory ) {
+		if ( isEmpty( visibleBlocksByCategory ) ) {
```

Comment:
If the only usage of this logic is when searching it may make sense to make this verification in line 358 where we call render categories as it makes things more explicit.

Reply:
This version with lodash cond is nice, but it is not obvious to read as you pointed out. The other possibility would save getVisibleBlocksByCategory in a const before the render but this way we may be computing it without need. I did not think about the complication because of being inline. It looks like our options have their downsides are not better than the current version we have, so I think we can ignore this detail and merge as it is right now.